### DISCUSSION TITLE
HELP: how to create a quotient of ZZ[x] or ZZ[x,y]?

### DISCUSSION THREAD

**Discussion by JohnAAbbott** [oscar-system/Oscar.jl#2916] [Q&A]
I wanted to create the quotient ring `ZZ[x]/ideal(6,(x+1)*(x+2)*(x+3))` but failed to find the correct approach in Oscar:
```
P,alpha = polynomial_ring(ZZ, "alpha");
I = ideal(P, [P(6),(alpha+1)*(alpha+2)*(alpha+3)]);  # <--- FAILS
k = quo(P,I);
```
I also tried another method:
```
ZZ6,_ = quo(ZZ,6);
P,alpha = polynomial_ring(ZZ6, "alpha");
I = ideal(P, [(alpha+1)*(alpha+2)*(alpha+3)]);  # <--- FAILS
```

What is the correct way to proceed?

**Comment by thofma** [comment]
It does not exist.

**Reply by thofma** [reply]
Or try `residue_ring(P, (alpha+1)*(alpha+2)*(alpha+3))` if it is a principal ideal.

**Reply by JohnAAbbott** [reply]
Thanks: that is a useful short-cut for principal ideals (but my ideal is not principal).

**Comment by ederc** [comment]
Well, if you define it as multivariate ring it works.

```
P,(alpha,) = polynomial_ring(ZZ, ["alpha"]);
```

**Reply by JohnAAbbott** [reply]
OK, thanks.  That works also for me... very odd that it baulks if I use the "univariate" constructor.  I suppose that makes it a bug.

**Reply by thofma** [reply]
Sounds more like a feature request.

**Reply by fieker** [reply]
or do it in 2 steps as a quotient of Z/6Z[t] - that can be constructed. We don't have ideals in ZZ[x]
